docs/web/concepts-key-concepts.mdx

```diff
@@ -0,0 +1,54 @@
+---
+title: Key Concepts
+description: Geocortex Web - Learn about key concepts of the Geocortex Web Framework
+hide_table_of_contents: true
+---
+
+import DesignerCallout from "./snippets/designer-callout.mdx";
+
+## Applications
+
+The core element of Geocortex Web is an **Application**. Applications are composed of three main pieces:
+
+#### 1. Layout XML that represents the visual layout of the application.
+
+```xml
+<layout xmlns="https://geocortex.com/layout/v1">
+    <map id="map-primary" config="default" />
+</layout>
+```
+
+An **layout** is composed of **components**, such as a `<map>`, which are the conceptual building blocks of an application. Components can define their own UI, consume configuration, and can be nested and interact with other components.
+
+Every component can have a `id` and `config` attribute; these two attributes are key to linking the layout to the **app config**.
+
+#### 2. App Config JSON that configures the application.
+
+```json
+{
```

Comment:
The schemaVersion requirement doesn't seem to be enforced, nor does it seem to break anything if it is removed. Why is it neccesary?

Reply:
It's equivalent to `<?xml version="1.0" ?>` in XML, or `DOCTYPE` in HTML, both of which are mandatory (well, in HTML5 and up). It tells us which version of the syntax you are using. Having that is important since it gives us an opportunity to change the spec later if needed. Making it optional would mean that we'd have apps in the wild with unknown schema version. We'd have to assume (wrongly, in some cases) the latest version, or else use other heuristics to try and guess. This would defeat the point of having this property; it either needs to be mandatory or not exist at all. Make sense? You're right that it's not currently enforced at runtime, but it's non-optional in the typings for `AppConfig`.

Reply:
Updated the documentation around the schema properties in the advanced config to reflect this information.

Ensured that every example that represents a full or partial app config has the schema version property, in both web and mobile

Reply:
Great, thanks.